insights/parsers/ssh.py

```diff
@@ -176,6 +179,52 @@ def get_line(self, keyword, default=''):
                 keyword=keyword, value=default
             )
 
+    def does_line_use_plus(self, keyword):
+        """
+        (union[bool, None]): Get the line with the last declarations of this
+        keyword in the configuration file and returns whether the "+" option
+        syntax is used.
+
+        A "+" before the list of values denotes that the values are appended to
+        the openssh defaults for the particular keyword.
+
+        Returns True if the "+" is used, False if a line with the keyword was
+        found but it doesn't use the "+" or None if such a line doesn't exist.
+
+        Reasoning for the implementation:
+
+        * The "+" means "added to the defaults".
+        * The defaults depend on the particular openssh-server version and
+          the parser doesn't know the version.
+        * Therefore, it is infeasible to add the evaluation logic for "+"
+          into `get_values()`.
+        * Adding the logic into a combiner would mean a requirement that
+          the combiner has a complete database of all defaults in all
+          openssh-server version - infeasible again.
+        * Not every keyword allows the use of "+" - it wouldn't make sense
+          to parse "+" into `KeyValue` as it would make meaningless parsing
+          for some options and meaningful for others. Building a database
+          which options in which openssh-server versions support it or not
+          would be infeasible.
+        * The way chosen as the most sensible is this - `does_line_use_plus()`
+          used selectively by a rule for those options that support it, and
+          it is up to the developer of such a rule to check it for those
+          options manually.
+
+        Parameters:
+            keyword(str): Keyword to find
+        """
+        lines = self.get(keyword)
+        if lines:
+            found_line = lines[-1].line
+            match = PLUS_PATTERN.match(found_line)
+            if match:
```

Comment:
`if thing return True else return False` should always IMO be just `return thing`.

Personally, I'd put named groups in the regex and return the group dictionary (via `match.groupdict()`).  But if for some reason you want to make it harder for your users, then just `return bool(match)` or `return match is not None`.

Reply:
@PaulWay I agree with what you have said. Just a small note, we usually do not have "your users" for our parsers. We usually write them for our rules only, so whatever difficulty Jakub will make in this parser method quite probably will only he suffer 😄.

Reply:
@PaulWay As to your first sentence, you are right. I wanted to put the emphasis that I want to return exactly `True` or `False` no matter how regex is implemented in any version of Python that the framework happens to run on. The correct way here would then be to `return bool(match)` if I understand you correctly. I will change it that way.